pkg/snapstore/snapshot_test.go

```diff
@@ -88,181 +88,121 @@ var _ = Describe("Snapshot", func() {
-					Expect(snapList[i].StartRevision).Should(BeNumerically(">", snapList[i-1].StartRevision))
-					Expect(snapList[i]).Should(Equal(expectedSnapList[i]))
-				}
-			})
-
-			It("sorts snapshot in preference order of snap directory creation time followed by snapshot creation time and then start revision", func() {
-				interval := int64(5)
-				now := time.Now().Unix()
-				snapdir1 := fmt.Sprintf("Backup-%d", now)
-				snapdir2 := fmt.Sprintf("Backup-%d", now+2*interval)
-				snap1 := Snapshot{
-					CreatedOn:     time.Unix(now, 0).UTC(),
-					StartRevision: 1001,
-					LastRevision:  1050,
-					Kind:          SnapshotKindDelta,
-					SnapDir:       snapdir1,
-				}
-				snap2 := Snapshot{
-					CreatedOn:     time.Unix(now+1*interval, 0).UTC(),
-					StartRevision: 1051,
-					LastRevision:  1200,
-					Kind:          SnapshotKindDelta,
-					SnapDir:       snapdir1,
-				}
-				snap3 := Snapshot{
-					CreatedOn:     time.Unix(now+1*interval, 0).UTC(),
-					StartRevision: 1201,
-					LastRevision:  1500,
-					Kind:          SnapshotKindDelta,
-					SnapDir:       snapdir1,
-				}
-				snap4 := Snapshot{
-					CreatedOn:     time.Unix(now+2*interval, 0).UTC(),
-					StartRevision: 1501,
-					LastRevision:  2000,
-					Kind:          SnapshotKindDelta,
-					SnapDir:       snapdir2,
-				}
-				snap5 := Snapshot{
-					CreatedOn:     time.Unix(now+2*interval, 0).UTC(),
-					StartRevision: 2001,
-					LastRevision:  2150,
-					Kind:          SnapshotKindDelta,
-					SnapDir:       snapdir2,
-				}
-				snap6 := Snapshot{
-					CreatedOn:     time.Unix(now+3*interval, 0).UTC(),
-					StartRevision: 2151,
-					LastRevision:  2160,
-					Kind:          SnapshotKindDelta,
-					SnapDir:       snapdir1,
-				}
-				snapList := SnapList{&snap5, &snap2, &snap1, &snap4, &snap6, &snap3}
-				sort.Sort(snapList)
-				expectedSnapList := SnapList{&snap1, &snap2, &snap3, &snap6, &snap4, &snap5}
+				expectedSnapList := brtypes.SnapList{&snap1, &snap2, &snap3, &snap4, &snap5, &snap6}
 				for i := 1; i < len(snapList); i++ {
+					Expect(snapList[i].LastRevision).Should(BeNumerically(">", snapList[i-1].LastRevision))
 					Expect(snapList[i]).Should(Equal(expectedSnapList[i]))
 				}
 			})
 		})
 
 		Context("given a snapshot", func() {
 			now := time.Now().Unix()
-			snap1 := Snapshot{
+			snap1 := brtypes.Snapshot{
 				CreatedOn:     time.Unix(now, 0).UTC(),
 				StartRevision: 0,
 				LastRevision:  2088,
-				Kind:          SnapshotKindFull,
+				Kind:          brtypes.SnapshotKindFull,
 			}
 			It("generates snapshot name ", func() {
 				snap1.GenerateSnapshotName()
 				Expect(snap1.SnapName).Should(Equal(fmt.Sprintf("Full-00000000-00002088-%08d", now)))
 			})
-			It("generates snapshot directory name ", func() {
-				snap1.GenerateSnapshotDirectory()
-				Expect(snap1.SnapDir).Should(Equal(fmt.Sprintf("Backup-%08d", now)))
-			})
 		})
 	})
 
 	Describe("Parse Snapshot name", func() {
```

Comment:
Can you please add the cases for with as well as without prefix? v1 and v2 if it makes a diffference.

Reply:
As per the out of band discussion, it is addressed.